docs/platform-integration/storage/file-system-helpers.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "File system helpers"
+description: "Learn how to use the .NET MAUI FileSystem class in the Microsoft.Maui.Storage namespace. This class contains helper methods that access the application's cache and data directories, and helps open files in the app package."
+ms.date: 05/13/2022
+no-loc: ["Microsoft.Maui", "Microsoft.Maui.Storage", "FileSystem"]
+---
+
+# File system helpers
+
+This article describes how you can use the .NET Multi-platform App UI (.NET MAUI) `IFileSystem` interface. This interface provides helper methods that access the app's cache and data directories, and helps access files in the app package. The `IFileSystem` interface is exposed through the `FileSystem.Current` property.
+
+[!INCLUDE [docs under construction](~/includes/preview-note.md)]
+
+The `FileSystem` and `IFileSystem` types are available in the `Microsoft.Maui.Storage` namespace.
+
+## Using file system helpers
+
+Each operating system will have unique paths to the app cache and app data directories. The `IFileSystem` interface provides a cross-platform API for accessing these directory paths.
+
+### Cache directory
+
+To get the application's directory to store **cache data**. Cache data can be used for any data that needs to persist longer than temporary data, but shouldn't be data that is required to operate the app, as the operating system may clear this storage.
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="filesys_cache":::
+
+### App data directory
+
+To get the app's top-level directory for any files that aren't user data files. These files are backed up with the operating system syncing framework.
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="filesys_appdata":::
+
+## Bundled files
+
+To open a file that is bundled into the app package, use the `OpenAppPackageFileAsync` method and pass the file name. This method returns an <xref:System.IO.Stream> representing the file contents. The following example demonstrates using a method to read the text contents of a file:
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="filesys_readtxtfile":::
+
+## Platform differences
+
+This section describes the platform-specific differences with the file system helpers.
+
+<!-- markdownlint-disable MD025 -->
+# [Android](#tab/android)
+
+- `FileSystem.CacheDirectory`\
+Returns the [CacheDir](https://developer.android.com/reference/android/content/Context.html#getCacheDir) of the current context.
+
+- `FileSystem.AppDataDirectory`\
+Returns the [FilesDir](https://developer.android.com/reference/android/content/Context.html#getFilesDir) of the current context, which are backed up using [Auto Backup](https://developer.android.com/guide/topics/data/autobackup.html) starting on API 23 and above.
+
+- `FileSystem.OpenAppPackageFileAsync`\
+Add any file into the _Assets_ folder in the Android project, and mark the **Build Action** as **AndroidAsset** to use it with `OpenAppPackageFileAsync`.
```

Comment:
This is old text. It may still be correct. But with MAUI Android asset files now live in the single project, and should have a build action of MauiAsset. So I'd update the text to say that files that are marked with the **MauiAsset** build action, that live in *Platform\Resources\Raw* folder, will work with this method.

Note: that folder isn't in the templates until the GA build.